### PR TITLE
Trusted Device Encryption feature

### PR DESCRIPTION
**Pull request by trmartin4**
## Type of change

```
- [ ] Bug fix
- [X] New feature development
- [ ] Tech debt (refactoring, code cleanup, dependency upgrades, etc)
- [ ] Build/deploy pipeline (DevOps)
- [ ] Other
```

## Objective

This is the merge of the TDE feature branch into master. Each individual PR on the branch has been reviewed.

## Code changes

See individual PRs for context.

## Before you submit

- Please check for formatting errors (`dotnet format --verify-no-changes`) (required)
- If making database changes - make sure you also update Entity Framework queries and/or migrations
- Please add **unit tests** where it makes sense to do so (encouraged but not required)
- If this change requires a **documentation update** - notify the documentation team
- If this change has particular **deployment requirements** - notify the DevOps team

### CODE REVIEW

File: src/Core/Auth/Services/Implementations/AuthRequestService.cs [hunk 1]
```diff
@@ -91,6 +103,42 @@ public async Task<AuthRequest> CreateAuthRequestAsync(AuthRequestCreateRequestMo
             }
         }
 
+        // AdminApproval requests require correlating the user and their organization
+        if (model.Type == AuthRequestType.AdminApproval)
+        {
+            // TODO: When single org policy is turned on we should query for only a single organization from the current user
```
Comment (JaredSnider-Bitwarden) [inline src/Core/Auth/Services/Implementations/AuthRequestService.cs:109]: @trmartin4 do we have a tech debt item for cleaning this up?

Comment (trmartin4) [reply src/Core/Auth/Services/Implementations/AuthRequestService.cs:109]: Yes! https://bitwarden.atlassian.net/browse/PM-3183

Comment (JaredSnider-Bitwarden) [reply src/Core/Auth/Services/Implementations/AuthRequestService.cs:109]: TY!

File: src/Core/Services/Implementations/UserService.cs [hunk 2]
```diff
@@ -1455,9 +1455,9 @@ public async Task SendOTPAsync(User user)
             throw new BadRequestException("No user email.");
         }
 
-        if (!user.UsesKeyConnector)
+        if (user.MasterPassword != null)
```
Comment (JaredSnider-Bitwarden) [inline src/Core/Services/Implementations/UserService.cs:1458]: @justindbaur , should this be using the `user.hasMasterPassword()` method?

**Review by JaredSnider-Bitwarden — DISMISSED** [review]
🎉  Looks good to go! All PRs have been individually reviewed and I've done a cursory re-review of this code.  🎉

**Review by jlf0dev — APPROVED** [review]

**Pull request merged**